### PR TITLE
Report stacktraces for remote_exec() 

### LINKED ISSUE #101: Tracebacks missing when using Gateway.remote_exec()
Hi.

If you call `remote_exec(mymodule)` and then an exception is raised, we get something like this

```
execnet.gateway_base.RemoteError: Traceback (most recent call last):
  File "<string>", line 1073, in executetask
  File "<remote exec>", line 79, in function1
  File "<remote exec>", line 77, in function2
ValueError: 'some error'
```
which is quite annoying to debug in a big application, as you have to hunt down where `remote_exec()` is called in the code, see which module was invoked, open the file and jump to the line manually.

The fix is simple. The proper file path just needs to be passed to the slave.

### PR DESCRIPTION
**Pull request by joaoe**
Fix #101

### CODE REVIEW

File: execnet/gateway.py [hunk 1]
```diff
@@ -111,22 +110,27 @@ def remote_exec(self, source, **kwargs):
             executing code.
         """
         call_name = None
+        file_name = None
         if isinstance(source, types.ModuleType):
-            linecache.updatecache(inspect.getsourcefile(source))
```
Comment (nicoddemus) [inline execnet/gateway.py]: Any reason why you dropped `linecache.updatecache`?

**Comment by nicoddemus** [comment]
Besides my comment, this looks good to me. 👍

File: execnet/gateway.py (hunk 1 above)
Comment (joaoe) [reply execnet/gateway.py]: Hi.

The `linecache.updatecache()` now runs in the slave process, so it saves some cpu cycles in the main process, will pin less stuff in memory, and is less stuff to pickle and pipe through to the slave. So, a small optimization.

Comment (nicoddemus) [reply execnet/gateway.py]: I see, thanks!

**Comment by nicoddemus** [comment]
Thanks again @joaoe and sorry for the delay on this! 👍

**Comment by nicoddemus** [comment]
1.7.0 released. 👍

**Comment by joaoe** [comment]
Excellent. Thank you :)

**Pull request merged**